Improvement and optimization

pavelpalonko opened the pull request.

pavelpalonko committed:
test

pavelpalonko committed:
fixeds

pavelpalonko committed:
fix name eventFunc

pavelpalonko committed:
feature

added input validation

pavelpalonko committed:
refactor

changed the style of functions to declarative

max-mykhailenko commented on src/common/hooks/useMatrixCalculations.tsx:
```diff
@@ -1,38 +1,31 @@
-    }
-
-    return matrix
+   return Array.from({length: rowMatrix}, () => (Array.from({length: colMatrix}, () => ({id: generetorId(), amount: randomInt(100, 999)}))))
```
Норм

max-mykhailenko commented on src/common/hooks/useMatrixCalculations.tsx:
```diff
@@ -1,38 +1,31 @@
     const matrixSum: MatrixDerivedProperties = { rowSumValues: [], columnAverageValues: [] }
 
-    for (let row of matrix) {
+    matrix.forEach(row => {
```
Reduce?

max-mykhailenko commented on src/common/hooks/useMatrixCalculations.tsx:
```diff
@@ -1,38 +1,31 @@
-      matrixSum.columnAverageValues.push(Math.round(sum / rowMatrix))
-    }
+    matrixSum.columnAverageValues = matrix.reduce((sum, row) => {
+      row.forEach((elem, index) => {
```
Reduce?

max-mykhailenko commented on src/components/InputNumber/InputNumber.tsx:
```diff
@@ -1,22 +1,27 @@
+  increment: MouseEventHandler<HTMLButtonElement>
+  decrement: MouseEventHandler<HTMLButtonElement>
+  matrixSize: InitialParameters
+  onChangeValue: Function
```
На самом деле function это тот же самый any, то есть нужно прописать точнее

max-mykhailenko commented on src/components/MatrixTable/MatrixTable.tsx:
```diff
@@ -53,34 +53,27 @@ const MatrixTable = ({ matrixSize, setMatrixSize }: MatrixTableProps ) => {
-    if ((event.target as HTMLElement).dataset.id) {
-      setClosestCells([...cellsAmount(event, matrix, matrixSize.X)])
-    }
+  const onHighlightCells = (currentCell: MatrixCell) => {
```
UseCallback?

max-mykhailenko commented on src/components/MatrixTableDisplayer/MatrixTableDisplayer.tsx:
```diff
@@ -31,16 +30,20 @@ const MatrixTableDisplayer = ({ matrix, closestCells, onMouseOutEvent, deleteRow
+      <tbody>
         {
-          matrix.map((rowMarix: MatrixRow, index: number) => (
+          matrix.map((rowMarix: MatrixRow, index: any) => (
```
Any?

max-mykhailenko commented on src/components/MatrixTableDisplayer/MatrixTableDisplayer.tsx:
```diff
@@ -31,16 +30,20 @@ const MatrixTableDisplayer = ({ matrix, closestCells, onMouseOutEvent, deleteRow
         {
-          matrix.map((rowMarix: MatrixRow, index: number) => (
+          matrix.map((rowMarix: MatrixRow, index: any) => (
               <tr key={index}>
```
А использовать индексы это плохая практика, производительность потом не вытянешь. Начинай думать про структуры данных и что можно использовать чтобы получить тут уникальные идентификаторы

max-mykhailenko replied:
А ще можна подумати про те, а чи не зробити таблицю просто стилями, тоді цикл однорівневий :)

max-mykhailenko commented on src/components/MatrixTableDisplayer/MatrixTableDisplayer.tsx:
```diff
@@ -31,16 +30,20 @@ const MatrixTableDisplayer = ({ matrix, closestCells, onMouseOutEvent, deleteRow
+                  onMouseOver={() => highlightCells(cell)} 
+                  onMouseOut={() => removeHighlightsCells()} 
+                  key={cell.id} 
+                  data-id={cell.id} 
```
Еще нужно?

pavelpalonko committed:
refactor

the display of the matrix through styles has been redone, the logic for calculating matrix derivatives has been redone, unique IDs have been added for matrix derivatives.

max-mykhailenko commented on src/common/hooks/useMatrixCalculations.tsx:
```diff
@@ -3,33 +3,49 @@ import { Matrix, MatrixDerivedProperties } from "../../models/matrix.models"
+  }
+
+  const addRowMatrix = (
+    matrix: any,
```
any? це чому ж? Тут можна прописати чіткий тип даних

max-mykhailenko commented on src/common/hooks/useMatrixCalculations.tsx:
```diff
@@ -3,33 +3,49 @@ import { Matrix, MatrixDerivedProperties } from "../../models/matrix.models"
 
-   return Array.from({length: rowMatrix}, () => (Array.from({length: colMatrix}, () => ({id: generetorId(), amount: randomInt(100, 999)}))))
-  
+    return Array.from({ length: 1 }, () => (Array.from({ length: colMatrix },
```
Array.from({ length: 1 }, () => (Array.from({ length: colMatrix }
ця дія часто повторюється, схоже що це може бути окрема функція

max-mykhailenko commented on src/common/hooks/useMatrixCalculations.tsx:
```diff
@@ -3,33 +3,49 @@ import { Matrix, MatrixDerivedProperties } from "../../models/matrix.models"
 
-  const calculateMatrix = useCallback( (rowMatrix: number, colMatrix: number, matrix: Matrix,) => {
-    
+  const calculateMatrix = useCallback((matrix: Matrix) => {
```
А навіщо тобі calculateMatrix як функція, а не просто значення, яке загорнуте в memo?

max-mykhailenko reviewed (approved):
Після невеликих правок можеш рухатись далі та тестувати швидкодію

pavelpalonko committed:
refactor

pavelpalonko committed:
fixed

added cell memoization
